File: python/1423.py
from typing import List

class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        points = [0] * (k + 1)
        l, r = 0, 0
        n = len(cardPoints)
        for i in range(k):
            l += cardPoints[i]
            r += cardPoints[-1-i]
            points[i+1] += l
            points[-2-i] += r
        return max(points)

File: go/1423.go
package main

func maxScore(cardPoints []int, k int) int {
	n := len(cardPoints)
	points := make([]int, k+1)
	left := 0
	right := 0
	for i := 0; i < k; i++ {
		left += cardPoints[i]
		right += cardPoints[n-1-i]
		points[i+1] += left
		points[k-1-i] += right
	}
	ans := 0
	for _, point := range points {
		if point > ans {
			ans = point
		}
	}
	return ans
}
